### PR TITLE
sig-cluster-lifecycle: amend the SIG charter

### PR DESCRIPTION
**Pull request by neolit123**
- [1] Remove ownership of the upgrade / downgrade testing of k8s.
Now owned by SIG Cloud Provider / GCP as it only supports
GCP clusters. See:
https://github.com/kubernetes/kubernetes/tree/master/test/e2e/cloud/gcp
- [2] Remove the "lock-step" coordination with SIG-Release. Historically
the release waited for kubeadm to update its default k8s versions
constants after a "stable-1.x" label was created. No longer needed,
as kubeadm defaults to "stable-1":
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta2/defaults.go#L36
- [3] Add sig-scheduling mention under "Out of scope".
- [4] Add sig-apps mention under "Out of scope".
- [5] Add note about release "informing/blocking" test jobs.

**Which issue(s) this PR fixes**:
NONE

/sig cluster-lifecycle
/priority important-longterm
/kind cleanup

as per:
https://github.com/kubernetes/community/blob/2915adc28cba40f74f0c1ba574c73509f826c7b9/committee-steering/governance/README.md#steps-to-update-an-existing-sig-charter

i think this falls under a "Minor update".

### CODE REVIEW

File: sig-cluster-lifecycle/charter.md [hunk 1]
```diff
@@ -22,9 +22,7 @@ The following topics fall under ownership of this SIG:
 
 #### Cross-cutting and Externally Facing Processes
 
-- This SIG works closely with SIG Release during the end of the cycle, because of how the release of kubeadm is currently managed. The process requires a lock-step coordination which is outlined [here](https://github.com/kubernetes/kubeadm/blob/master/docs/release-cycle.md).
 - The SIG recommends and verifies compatibility of critical cluster add-ons for networking, network policy, service discovery, etc. The SIG maintains the health check of container images for some add-ons that are required for cluster bootstrapping. While the SIG could provide support to users that have add-on related issues, the SIG can decide to delegate issues to the add-on maintainers or other SIGs.
-- The SIG owns the framework for upgrade / downgrade testing of Kubernetes. However, the SIG does not own the individual tests for each feature. While the SIG is often involved in triaging and debugging upgrade or downgrade test failures, the SIG will normally delegate issues to the feature owner for the failing test.
```
Comment (neolit123) [inline sig-cluster-lifecycle/charter.md:27]: [1]

File: sig-cluster-lifecycle/charter.md [hunk 2]
```diff
@@ -22,9 +22,7 @@ The following topics fall under ownership of this SIG:
 
 #### Cross-cutting and Externally Facing Processes
 
-- This SIG works closely with SIG Release during the end of the cycle, because of how the release of kubeadm is currently managed. The process requires a lock-step coordination which is outlined [here](https://github.com/kubernetes/kubeadm/blob/master/docs/release-cycle.md).
```
Comment (neolit123) [inline sig-cluster-lifecycle/charter.md:25]: [2]

File: sig-cluster-lifecycle/charter.md [hunk 3]
```diff
@@ -34,7 +32,8 @@ The following topics fall under ownership of this SIG:
 - User interface, or user experience, issues other than cluster bootstrapping or management (see [sig-ui](../sig-ui) and  [sig-cli](../sig-cli)).
 - Node related issues (see [sig-node](../sig-node)).
 - Kubernetes control plane issues:
-   - Control plane component related issues (see [sig-api-machinery](../sig-api-machinery)).
+   - Control plane component related issues (see [sig-api-machinery](../sig-api-machinery) and [sig-scheduling](../sig-scheduling)).
```
Comment (neolit123) [inline sig-cluster-lifecycle/charter.md:35]: [3]

File: sig-cluster-lifecycle/charter.md [hunk 4]
```diff
@@ -34,7 +32,8 @@ The following topics fall under ownership of this SIG:
...
+- Deployment and lifecycle issues related to user application deployments (see [sig-apps](../sig-apps)).
```
Comment (neolit123) [inline sig-cluster-lifecycle/charter.md:36]: [4]

File: sig-cluster-lifecycle/charter.md [hunk 5]
```diff
@@ -51,7 +50,7 @@ and opts-in to updates and modifications to [sig-governance].
 
 ### Deviations from [sig-governance]
 
-- As SIG cluster lifecycle contains a number of subprojects, the SIG has empowered subproject leads with a number of additional responsibilities, including but not limited to:
+- As SIG Cluster Lifecycle contains a number of subprojects, the SIG has empowered subproject leads with a number of additional responsibilities, including but not limited to:
```
Comment (neolit123) [inline sig-cluster-lifecycle/charter.md:53]: upper case the SIG name.

File: sig-cluster-lifecycle/charter.md [hunk 6]
```diff
@@ -9,7 +9,7 @@ SIG Cluster Lifecycle’s objective is to simplify creation, configuration, upgr
 The following topics fall under ownership of this SIG:
 
 - Improving the Kubernetes user experience for cluster administration.
-- Tools that assist in the creation, configuration, upgrade, downgrade, and teardown of Kubernetes control plane components. 
+- Tools that assist in the creation, configuration, upgrade, downgrade, and teardown of Kubernetes control plane components.
```
Comment (neolit123) [inline sig-cluster-lifecycle/charter.md:12]: white space cleanup.

**Comment by cblecker** [comment]
@neolit123 To quote governance.md:
> For significant changes, or **any changes that could impact other SIGs**

Scope changes, and coordination with other sigs fall under the latter point. It may be just a formality, but please do engage steering on this. Thank you!

/hold
/committee steering

**Comment by neolit123** [comment]
> but please do engage steering on this

thanks for the heads up @cblecker. email sent.

File: sig-cluster-lifecycle/charter.md (hunk 2 above)
Comment (timothysc) [reply sig-cluster-lifecycle/charter.md:25]: I think we should may a statement about published release informing jobs.

**Comment by detiber** [comment]
+1 to these changes

File: sig-cluster-lifecycle/charter.md (hunk 2 above)
Comment (neolit123) [reply sig-cluster-lifecycle/charter.md:25]: other SIGs are also involved with SIG Release in that regard, but i guess it makes sense adding a note for it.

Comment (neolit123) [reply sig-cluster-lifecycle/charter.md:25]: added as [5]

File: sig-cluster-lifecycle/charter.md [hunk 7]
```diff
@@ -22,19 +22,19 @@ The following topics fall under ownership of this SIG:
...
-- The SIG owns the framework for upgrade / downgrade testing of Kubernetes. However, the SIG does not own the individual tests for each feature. While the SIG is often involved in triaging and debugging upgrade or downgrade test failures, the SIG will normally delegate issues to the feature owner for the failing test.
 - The SIG collaborates regularly with SIG Auth in an effort to follow best practices in order to promote secure default clusters.
 - The SIG co-owns cloud provider specific code related to cluster and machine provisioning with the respective SIGs for each cloud provider but does not own the cloud controller manager or any other provider specific code.
+- The SIG ensures that the Kubernetes "release informing" and "release blocking" E2E test jobs that it maintains are in good health.
```
Comment (neolit123) [inline sig-cluster-lifecycle/charter.md:28]: [5]

File: sig-cluster-lifecycle/charter.md (hunk 3 above)
Comment (spiffxp) [reply sig-cluster-lifecycle/charter.md:36]: does this imply that sig-api-machinery owns https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-component-upgrade-order

Comment (neolit123) [reply sig-cluster-lifecycle/charter.md:36]: > Control plane component related issues

refers to bugs around kube-apiserver and kube-controller-manager owned by SIG API Machinery and kube-scheduler owned by SIG Scheduling.

> does this imply that sig-api-machinery owns

i'd assume that given the upgrade order guideline is cutting across multiple core components that the real owner of the should be SIG Architecture with technical guidance by the component owning SIGs.

> Tools that assist in the creation, configuration, upgrade, downgrade, and teardown of Kubernetes control plane components.

with the order in place tools created by SIG Cluster Lifecycle should follow the guideline, as much as possible, or at least this is my interpretation.

File: sig-cluster-lifecycle/charter.md (hunk 1 above)
Comment (neolit123) [reply sig-cluster-lifecycle/charter.md:27]: > Agree on not owning the framework, but I wonder if this leaves the project without ownership of verifying pre-conditions / post-conditions that are expected to hold for kubernetes across cluster providers

my take about the upgrade framework is the following, the variety of Kubernetes component topologies and its modularity renders the chances of a single upgrade framework to work across cluster providers unlikely, unless we start standardizing topology.

the entry that is being removed here is something that works for [GCE/GKE](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/cloud/gcp/cluster_upgrade.go#L614-L617), yet i cannot confirm if it currently supports pre/post condition testing across cluster providers.

> Put another way: there are some upgrade/skew tests in release-informing based on kind. Are those representative of how we would expect all kubernetes upgrades/skews to perform?

i'm surprised that kubeadm/kinder jobs are the only upgrade jobs in `release-*` dashboards at this point. the kubeadm upgrades do not strictly follow the upgrade/skew document, due to topology differences. but they are close on the critical parts.

next to cluster-api upgrades to be added as release-* signal, in my opinion cloud providers need to step up and start doing federated upgrade testing at minimum.

**Comment by cblecker** [comment]
I share the same concerns as @spiffxp here: https://github.com/kubernetes/community/pull/4755#pullrequestreview-405216799

**Comment by neolit123** [comment]
hi, @spiffxp @cblecker 

are you suggesting the addition of a new item in the charter that touches on the following?
>  who now owns/verifies supported upgrade/downgrade procedures/policy

my reply (and opinion) to that was here:
https://github.com/kubernetes/community/pull/4755#discussion_r419732695

> i'd assume that given the upgrade order guideline is cutting across multiple core components that the real owner of the should be SIG Architecture with technical guidance by the component owning SIGs.

in addition to that, while i don't think SIG Cluster Lifecycle should own the policy, the group can help with the verification procedure (which we are already doing partially).

hi, @liggitt as the author of:
https://github.com/kubernetes/website/commit/75ed07f7169f1b5e96e0684c86e6f4699eefd779

what is your take on the above discussion?

**Comment by timothysc** [comment]
> My only question is around who now owns/verifies supported upgrade/downgrade procedures/policy, which on the face of it sounds like it belongs in the scope of sig-cluster-lifecycle 

We support the upgrade and downgrade procedures for the tools we maintain, but policy has always been dictated to us since epoch.  I think policy should be on sig-release and sig-arch.

**Comment by neolit123** [comment]
hello, post the discussion in this [SIG Arch mailing list thread](https://groups.google.com/d/msgid/kubernetes-sig-architecture/CALOex7tjhMvmjb7_ZGDHL26eYEKwPrHPtzWydmKGyDtKejMGrw%40mail.gmail.com?utm_medium=email&utm_source=footer
)

do we have remaining arguments against the changes made for this charter?

**Comment by spiffxp** [comment]
/approve
/lgtm
/hold cancel
I opened https://github.com/kubernetes/community/issues/4839 as a placeholder to address my concerns re: upgrade/downgrade. I don't think this PR needs to be blocked on those concerns anymore
